bokehjs/src/coffee/models/renderers/glyph_renderer.coffee

```diff
@@ -157,8 +157,7 @@ export class GlyphRendererView extends RendererView
       else if selected['1d'].indices.length > 0
         selected = selected['1d'].indices
       else
-        selected = Object.keys(selected["2d"].indices)
-        selected = (parseInt(i) for i in selected)
+        selected = (parseInt(i) for i in Object.keys(selected["2d"].indices))
```

Comment:
yup. I spent 20 minutes trying to figure out why I couldn't get values I'd just written to this dict.